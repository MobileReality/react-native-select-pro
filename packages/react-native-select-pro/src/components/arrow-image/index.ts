export { ArrowImage } from './arrow-image';
