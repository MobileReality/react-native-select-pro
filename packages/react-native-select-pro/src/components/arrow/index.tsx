import type { ComponentProps } from 'react';
import React from 'react';
import type { ImageStyle } from 'react-native';
import { Animated, Image, Pressable, StyleSheet } from 'react-native';

import type { OptionalToRequired } from '../../helpers/types/optional-to-required';
import { useAnimation } from '../../hooks/use-animation';
import type { State } from '../../state/types';
import type { OnPressSelectControlType } from '../../types';
import type { Select } from '../select';

const arrowImage = require('./../../assets/icons/chevron-down.png');

type ArrowProps = OptionalToRequired<
    { onPressSelectControl: OnPressSelectControlType } & Pick<State, 'isOpened'> &
        Pick<
            ComponentProps<typeof Select>,
            | 'animation'
            | 'multiSelection'
            | 'disabled'
            | 'selectControlArrowImageStyle'
            | 'customSelectControlArrowIconSource'
        >
>;

export const Arrow = ({
    isOpened,
    disabled,
    animation,
    multiSelection,
    selectControlArrowImageStyle,
    customSelectControlArrowIconSource,
    onPressSelectControl,
}: ArrowProps) => {
    const rotateAnimation = useAnimation({ isOpened, animation });
    const accessibilityLabel = 'Arrow for opening dropdown';

    const resolveContent = () => {
        const arrowSource = customSelectControlArrowIconSource ?? arrowImage;

        if (rotateAnimation) {
            const rotate = rotateAnimation.interpolate({
                inputRange: [0, 1],
                outputRange: ['0deg', '180deg'],
            });

            return (
                <Animated.Image
                    source={arrowSource}
                    style={[
                        styles.arrowIcon,
                        { transform: [{ rotate }] },
                        selectControlArrowImageStyle,
                    ]}
                />
            );
        }

        return (
            <Image
                source={arrowSource}
                style={[
                    styles.arrowIcon,
                    isOpened ? styles.arrowIconOpened : styles.arrowIconClosed,
                    selectControlArrowImageStyle,
                ]}
            />
        );
    };
    const arrowContent = resolveContent();

    if (!multiSelection) {
        return arrowContent;
    }

    return (
        <Pressable
            accessibilityLabel={accessibilityLabel}
            onPress={disabled ? undefined : onPressSelectControl}
        >
            {arrowContent}
        </Pressable>
    );
};

type Styles = {
    arrowIcon: ImageStyle;
    arrowIconOpened: ImageStyle;
    arrowIconClosed: ImageStyle;
};

const styles = StyleSheet.create<Styles>({
    arrowIcon: {
        width: 25,
        height: 25,
        zIndex: -1,
    },
    arrowIconOpened: {
        transform: [{ rotate: '180deg' }],
    },
    arrowIconClosed: {
        transform: [{ rotate: '0deg' }],
    },
});
