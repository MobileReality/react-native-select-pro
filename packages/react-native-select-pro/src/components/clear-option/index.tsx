import React, { ComponentProps } from 'react';
import {
    Image,
    ImageStyle,
    Pressable,
    StyleSheet,
    ViewStyle,
} from 'react-native';

import type { OptionalToRequired } from '../../helpers/types/OptionalToRequired';
import type { SelectControl } from '../select-control';

type FromSelectControlProps = Pick<
    ComponentProps<typeof SelectControl>,
    | 'selectControlClearOptionA11yLabel'
    | 'disabled'
    | 'selectControlClearOptionButtonHitSlop'
    | 'selectControlClearOptionButtonStyle'
    | 'selectControlClearOptionImageStyle'
> & {
    onPressRemove: () => void;
};

type ClearOptionProps = OptionalToRequired<FromSelectControlProps>;

export const ClearOption = ({
    selectControlClearOptionA11yLabel,
    disabled,
    selectControlClearOptionButtonHitSlop,
    selectControlClearOptionButtonStyle,
    selectControlClearOptionImageStyle,
    onPressRemove,
}: ClearOptionProps) => {
    return (
        <Pressable
            accessibilityLabel={
                selectControlClearOptionA11yLabel || 'Clear a chosen option'
            }
            accessibilityRole="button"
            accessible={true}
            disabled={disabled}
            hitSlop={
                selectControlClearOptionButtonHitSlop
                    ? selectControlClearOptionButtonHitSlop
                    : { right: 3, left: 3 }
            }
            style={[styles.xIconWrapper, selectControlClearOptionButtonStyle]}
            onPress={onPressRemove}
        >
            <Image
                source={require('./../../assets/icons/x.png')}
                style={[styles.xIcon, selectControlClearOptionImageStyle]}
            />
        </Pressable>
    );
};

type Styles = {
    xIcon: ImageStyle;
    xIconWrapper: ViewStyle;
};

const styles = StyleSheet.create<Styles>({
    xIconWrapper: {
        height: '100%',
        justifyContent: 'center',
    },
    xIcon: {
        width: 20,
        height: 20,
        zIndex: 1,
    },
});
