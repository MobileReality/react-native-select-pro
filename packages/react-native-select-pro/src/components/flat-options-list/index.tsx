export { FlatOptionsList } from './flat-options-list';
