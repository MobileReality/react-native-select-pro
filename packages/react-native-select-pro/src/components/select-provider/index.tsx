import React, { ReactNode } from 'react';
import { PortalHost, PortalProvider } from '@gorhom/portal';

import { Portals } from '../../constants/portals';

type Props = {
    children: ReactNode;
};

const PortalHosts = () => {
    return (
        <>
            <PortalHost name={Portals.SelectOutsideWrapper} />
            <PortalHost name={Portals.Select} />
        </>
    );
};

export const SelectProvider = ({ children }: Props) => {
    return (
        <PortalProvider>
            {children}
            <PortalHosts />
        </PortalProvider>
    );
};

export const SelectModalProvider = ({ children }: Props) => {
    return (
        <>
            <PortalHosts />
            {children}
        </>
    );
};
