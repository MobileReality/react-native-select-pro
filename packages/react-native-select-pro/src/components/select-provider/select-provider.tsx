import type { ReactNode } from 'react';
import React, { createContext } from 'react';
import { PortalHost, PortalProvider } from '@gorhom/portal';

import { APPROX_STATUSBAR_HEIGHT, Portals } from '../../constants';

type Props = {
    children: ReactNode;
};

const PortalHosts = () => {
    return (
        <>
            <PortalHost name={Portals.Backdrop} />
            <PortalHost name={Portals.OptionsList} />
        </>
    );
};

export const SelectProvider = ({ children }: Props) => {
    return (
        <PortalProvider>
            {children}
            <PortalHosts />
        </PortalProvider>
    );
};

export const SelectModalContext = createContext(0);

export const SelectModalProvider = ({ children }: Props) => {
    return (
        <SelectModalContext.Provider value={APPROX_STATUSBAR_HEIGHT}>
            <PortalHosts />
            {children}
        </SelectModalContext.Provider>
    );
};
