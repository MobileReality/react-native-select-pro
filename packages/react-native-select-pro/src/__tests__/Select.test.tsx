import React from 'react';
import { MeasureOnSuccessCallback, Pressable, Text, View } from 'react-native';
import { fireEvent, render } from '@testing-library/react-native';

import { Select } from '../components/select';
import { SelectProvider } from '../components/select-provider';
import type { OptionComponentProps } from '../index';

const DATA = [
    {
        value: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
        label: 'First Option',
    },
    {
        value: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
        label: 'Second Option',
    },
    {
        value: '58694a0f-3da1-471f-bd96-145571e29d72',
        label: 'Third Option',
    },
];

const SEARCHABLE_DATA = [
    {
        value: 'test1',
        label: 'First test options',
    },
    {
        value: 'test2',
        label: 'Second test options',
    },
    {
        value: 'test3',
        label: 'Third test option',
    },
];

const searchPattern = (value: string) => `^${value}`;

jest.spyOn(View.prototype, 'measure').mockImplementation(
    (fn: MeasureOnSuccessCallback): void => {
        fn(0, 0, 45, 20, 0, 0);
    },
);

describe('Select', () => {
    it('should generate Select snapshot', () => {
        const wrapper = render(
            <SelectProvider>
                <Select options={DATA} />
            </SelectProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should open options menu after press select control', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const list = getByLabelText('Options list');
        expect(list).toBeTruthy();
    });

    /**
     * to achieve this you need to pass `clearable={true}` and should be selected option
     */
    it('should show clear selected option button', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select clearable={true} options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const optionPress = getByLabelText(`Choose ${DATA[0].label} option`);
        fireEvent.press(optionPress);

        const clearButton = getByLabelText('Clear a chosen option');
        expect(clearButton).toBeTruthy();
    });

    it('should close dropdown menu after pressed outside dropdown', () => {
        const { getByLabelText, queryByLabelText } = render(
            <SelectProvider>
                <Select options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const listWrapper = queryByLabelText('Options list');
        expect(listWrapper).toBeTruthy();

        const outside = getByLabelText('Close a dropdown from outside');
        fireEvent.press(outside);

        const listWrapper2 = queryByLabelText('Options list');
        expect(listWrapper2).toBeFalsy();
    });

    it('should select option with custom option component', () => {
        const MyCustomOption = ({
            onPressOption,
            option,
        }: OptionComponentProps) => {
            return (
                <Pressable accessibilityLabel="Option" onPress={onPressOption}>
                    <Text>{option.label}</Text>
                </Pressable>
            );
        };
        const { getByLabelText } = render(
            <SelectProvider>
                <Select
                    OptionComponent={(props) => <MyCustomOption {...props} />}
                    options={[DATA[0]]}
                />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const list = getByLabelText('Options list');
        expect(list).toBeTruthy();

        const option = getByLabelText('Option');
        fireEvent.press(option);
    });
});

describe('Select with searchable enabled', () => {
    it('should generate Select with searchable enabled snapshot', () => {
        const wrapper = render(
            <SelectProvider>
                <Select options={SEARCHABLE_DATA} searchable={true} />
            </SelectProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should open options menu after pressing Pressable in select control with searchable enabled', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select options={SEARCHABLE_DATA} searchable={true} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const list = getByLabelText('Options list');
        expect(list).toBeTruthy();
    });

    it('should NOT open options menu after pressing Pressable or Input in select control with searchable enabled and whole select disabled', () => {
        const onOpen = jest.fn();

        const { getByLabelText } = render(
            <SelectProvider>
                <Select
                    disabled={true}
                    options={SEARCHABLE_DATA}
                    searchable={true}
                    onDropdownOpened={onOpen}
                />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);
        expect(onOpen).not.toHaveBeenCalled();

        const testInput = getByLabelText('Place text');
        fireEvent.press(testInput);
        expect(onOpen).not.toHaveBeenCalled();
    });

    it('should open options menu after pressing Input in select control with searchable enabled', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select options={SEARCHABLE_DATA} searchable={true} />
            </SelectProvider>,
        );

        const open = getByLabelText('Place text');
        fireEvent.press(open);

        const list = getByLabelText('Options list');
        expect(list).toBeTruthy();
    });

    it('should type text in Input in select control with searchable enabled', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select options={SEARCHABLE_DATA} searchable={true} />
            </SelectProvider>,
        );

        const inputData = 'Testing messing';

        const input = getByLabelText('Place text');
        fireEvent.changeText(input, inputData);

        expect(input.props.value).toBe(inputData);
    });

    it('should not able to type text in Input in select control with searchable enabled but disabled', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select
                    disabled={true}
                    options={SEARCHABLE_DATA}
                    searchable={true}
                />
            </SelectProvider>,
        );

        const inputData = 'Testing messing';

        const input = getByLabelText('Place text');
        fireEvent.changeText(input, inputData);

        expect(input.props.value).toBe('');
    });

    it('should, while searchable enabled, search by label and default pattern', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select options={SEARCHABLE_DATA} searchable={true} />
            </SelectProvider>,
        );

        const firstInputData = 'Second test options';
        const secondInputData = 'Third test option';
        const thirdInputData = 'options';

        const input = getByLabelText('Place text');

        fireEvent.changeText(input, firstInputData);

        const list = getByLabelText('Options list');

        expect(list.props.data.length).toBe(1);

        fireEvent.changeText(input, secondInputData);
        expect(list.props.data.length).toBe(1);

        fireEvent.changeText(input, thirdInputData);
        expect(list.props.data.length).toBe(2);

        const valueData = 'test1';

        fireEvent.changeText(input, valueData);
        expect(list.props.data.length).toBe(0);
    });

    it('should, while searchable enabled, search by label and other patterns', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select
                    options={SEARCHABLE_DATA}
                    searchPattern={searchPattern}
                    searchable={true}
                />
            </SelectProvider>,
        );

        const firstInputData = 'Second';
        const secondInputData = 'option';
        const input = getByLabelText('Place text');

        fireEvent.changeText(input, firstInputData);

        const list = getByLabelText('Options list');

        expect(list.props.data.length).toBe(1);

        fireEvent.changeText(input, secondInputData);
        expect(list.props.data.length).toBe(0);
    });

    it('should, while searchable enabled, get back to previous value in select if clicked outside', () => {
        const { getByLabelText } = render(
            <SelectProvider>
                <Select options={SEARCHABLE_DATA} searchable={true} />
            </SelectProvider>,
        );

        const inputData = 'Second';
        const nextInputData = 'option';

        const selectOptionInputData = 'Second test options';

        const input = getByLabelText('Place text');

        fireEvent.changeText(input, inputData);

        const list = getByLabelText('Options list');

        expect(list.props.data.length).toBe(1);

        const option = getByLabelText(
            `Choose ${inputData} test options option`,
        );

        fireEvent.press(option);

        expect(input.props.value).toBe(selectOptionInputData);

        fireEvent.changeText(input, nextInputData);

        const listAgain = getByLabelText('Options list');
        expect(listAgain.props.data.length).toBeGreaterThan(1);
        expect(input.props.value).toBe(nextInputData);

        const outside = getByLabelText('Close a dropdown from outside');
        fireEvent.press(outside);

        expect(input.props.value).toBe(selectOptionInputData);
    });
});

describe('Select with custom left icon', () => {
    it('should generate Select with custom left icon snapshot', () => {
        const wrapper = render(
            <SelectProvider>
                <Select
                    customLeftIconSource={require('./assets/search.png')}
                    // eslint-disable-next-line react-native/no-inline-styles
                    customLeftIconStyles={{ height: 15, width: 15 }}
                    options={DATA}
                />
            </SelectProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Select with multi selection', () => {
    it('should generate Select with multi selection enabled snapshot', () => {
        const wrapper = render(
            <SelectProvider>
                <Select multiSelection={true} options={DATA} />
            </SelectProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should, while multiSelection enabled, click and open options', () => {
        const { getByLabelText, queryByLabelText } = render(
            <SelectProvider>
                <Select multiSelection={true} options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const listWrapper = queryByLabelText('Options list');
        expect(listWrapper).toBeTruthy();

        const outside = getByLabelText('Close a dropdown from outside');
        fireEvent.press(outside);

        const listWrapper2 = queryByLabelText('Options list');
        expect(listWrapper2).toBeFalsy();
    });

    it('should, while multiSelection enabled, click should NOT execute opening dropdown', () => {
        const { getByLabelText, queryByLabelText } = render(
            <SelectProvider>
                <Select multiSelection={true} options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const listWrapper = queryByLabelText('Options list');
        expect(listWrapper).toBeTruthy();

        const optionPress = getByLabelText(`Choose ${DATA[0].label} option`);
        fireEvent.press(optionPress);

        const optionSelected = getByLabelText(`${DATA[0].label} selected`);
        expect(optionSelected).toBeTruthy();

        const tryOpenAgain = getByLabelText('Open a dropdown');
        fireEvent.press(tryOpenAgain);

        const listWrapperShouldNotBeVisible = queryByLabelText('Options list');
        expect(listWrapperShouldNotBeVisible).toBeFalsy();
    });

    it('should, while multiSelection enabled, show, select and remove selected option', () => {
        const { getByLabelText, queryByLabelText } = render(
            <SelectProvider>
                <Select multiSelection={true} options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);

        const optionPress = getByLabelText(`Choose ${DATA[0].label} option`);
        fireEvent.press(optionPress);

        const optionSelected = getByLabelText(`${DATA[0].label} selected`);
        expect(optionSelected).toBeTruthy();

        const openAgain = getByLabelText('Arrow for opening dropdown');
        fireEvent.press(openAgain);

        const secondOptionPress = getByLabelText(
            `Choose ${DATA[1].label} option`,
        );
        fireEvent.press(secondOptionPress);

        const selectedSecondOption = getByLabelText(
            `${DATA[1].label} selected`,
        );
        expect(selectedSecondOption).toBeTruthy();

        fireEvent.press(optionSelected);

        const optionShouldNotExist = queryByLabelText(
            `${DATA[0].label} selected`,
        );
        expect(optionShouldNotExist).toBeFalsy();
    });
    it('should NOT open options menu after pressing Pressable in select with multi select enabled and whole select disabled', () => {
        const onOpen = jest.fn();

        const { getByLabelText } = render(
            <SelectProvider>
                <Select disabled={true} multiSelection={true} options={DATA} />
            </SelectProvider>,
        );

        const open = getByLabelText('Open a dropdown');
        fireEvent.press(open);
        expect(onOpen).not.toHaveBeenCalled();
    });
});
describe('Select with multi selection and searchable', () => {
    it('should generate Select with multi selection and searchable enabled snapshot', () => {
        const wrapper = render(
            <SelectProvider>
                <Select
                    multiSelection={true}
                    options={SEARCHABLE_DATA}
                    searchable={true}
                />
            </SelectProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should, while multiSelection and searchable enabled, click and open options', () => {
        const { getByLabelText, getByPlaceholderText } = render(
            <SelectProvider>
                <Select
                    multiSelection={true}
                    options={SEARCHABLE_DATA}
                    searchable={true}
                />
            </SelectProvider>,
        );

        const inputData = 'Second';
        const nextInputData = 'First';

        const input = getByLabelText('Place text');

        fireEvent.changeText(input, inputData);

        const list = getByLabelText('Options list');

        expect(list.props.data.length).toBe(1);

        const firstOption = getByLabelText(
            `Choose ${inputData} test options option`,
        );

        fireEvent.press(firstOption);

        const inputAgain = getByLabelText('Place text');

        expect(inputAgain.props.value).toBe('');

        fireEvent.changeText(inputAgain, nextInputData);
        expect(inputAgain.props.value).toBe(nextInputData);

        const listAgain = getByLabelText('Options list');

        expect(listAgain.props.data.length).toBe(1);

        const secondOption = getByLabelText(
            `Choose ${nextInputData} test options option`,
        );

        fireEvent.press(secondOption);

        const selectedFirstOption = getByLabelText(
            `${SEARCHABLE_DATA[1].label} selected`,
        );
        expect(selectedFirstOption).toBeTruthy();
        const selectedSecondOption = getByLabelText(
            `${SEARCHABLE_DATA[0].label} selected`,
        );
        expect(selectedSecondOption).toBeTruthy();

        const currentPlaceholderText = getByPlaceholderText('');
        expect(currentPlaceholderText).toBeTruthy();
    });
});
