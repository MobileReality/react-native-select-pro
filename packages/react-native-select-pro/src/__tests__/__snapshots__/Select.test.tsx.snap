// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should generate Select snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Open a dropdown"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {},
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            },
            Object {
              "paddingRight": 55,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 12,
                "textAlign": "left",
              },
              undefined,
              Object {
                "color": "#808080",
              },
            ]
          }
        >
          Select...
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "position": "absolute",
              "right": 8,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "packages/react-native-select-pro/src/assets/icons/chevron-down.png",
            }
          }
          style={
            Array [
              Object {
                "height": 25,
                "width": 25,
                "zIndex": -1,
              },
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Select with custom left icon should generate Select with custom left icon snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Open a dropdown"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {},
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingLeft": 8,
            },
            Object {
              "height": "100%",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "packages/react-native-select-pro/src/__tests__/assets/search.png",
            }
          }
          style={
            Object {
              "height": 15,
              "width": 15,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            },
            Object {
              "paddingRight": 55,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 12,
                "textAlign": "left",
              },
              undefined,
              Object {
                "color": "#808080",
              },
            ]
          }
        >
          Select...
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "position": "absolute",
              "right": 8,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "packages/react-native-select-pro/src/assets/icons/chevron-down.png",
            }
          }
          style={
            Array [
              Object {
                "height": 25,
                "width": 25,
                "zIndex": -1,
              },
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Select with multi selection and searchable should generate Select with multi selection and searchable enabled snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Open a dropdown"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {},
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            },
            Object {
              "paddingRight": 40,
            },
          ]
        }
      >
        <RCTScrollView
          horizontal={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <TextInput
              accessibilityLabel="Place text"
              editable={true}
              onChangeText={[Function]}
              onPressIn={[Function]}
              placeholder="Select..."
              placeholderTextColor="#808080"
              style={
                Array [
                  Object {
                    "fontSize": 12,
                  },
                  undefined,
                ]
              }
              textAlign="left"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "position": "absolute",
              "right": 8,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityLabel="Arrow for opening dropdown"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "packages/react-native-select-pro/src/assets/icons/chevron-down.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 25,
                  "width": 25,
                  "zIndex": -1,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Select with multi selection should generate Select with multi selection enabled snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Open a dropdown"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {},
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            },
            Object {
              "paddingRight": 40,
            },
          ]
        }
      >
        <RCTScrollView
          horizontal={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              accessibilityLabel="Placeholder in multi-select"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "display": "flex",
                    "justifyContent": "center",
                  },
                  undefined,
                  Object {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "fontSize": 12,
                      "textAlign": "left",
                    },
                    undefined,
                    Object {
                      "color": "#808080",
                    },
                  ]
                }
              >
                Select...
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "position": "absolute",
              "right": 8,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityLabel="Arrow for opening dropdown"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "packages/react-native-select-pro/src/assets/icons/chevron-down.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 25,
                  "width": 25,
                  "zIndex": -1,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Select with searchable enabled should generate Select with searchable enabled snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Open a dropdown"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {},
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            },
            Object {
              "paddingRight": 55,
            },
          ]
        }
      >
        <TextInput
          accessibilityLabel="Place text"
          editable={true}
          onChangeText={[Function]}
          onPressIn={[Function]}
          placeholder="Select..."
          placeholderTextColor="#808080"
          style={
            Array [
              Object {
                "fontSize": 12,
              },
              undefined,
            ]
          }
          textAlign="left"
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "position": "absolute",
              "right": 8,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "packages/react-native-select-pro/src/assets/icons/chevron-down.png",
            }
          }
          style={
            Array [
              Object {
                "height": 25,
                "width": 25,
                "zIndex": -1,
              },
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
